include/databaseFunctions_blog_pic.php

```diff
@@ -1,58 +1,250 @@
-					var key = prompt('Please enter delete key.', '');
-					if(key == delKey){
-						document.write(key);
-						document.write(delKey);
+			<form method='post'>
+		";
+		ShowTextField(true, 'username');
+		ShowPasswordField('password', 'password');
+		ShowPasswordField('confirm', 'confirm password');
+		ShowTextField(true, 'email');
+		echo"<input type='submit' name = 'create' value='Create Account'>
+			</form>";
+
+	}
+
+	function Login(){
+		$errors = array();
+		// var_dump($_REQUEST);
+		if(isset($_REQUEST['login'])){		//NOT PASSING THROUGH IF
+			// var_dump($_REQUEST);
+			$errors+=ValidateTextField('username', $errors);
+			$errors+=ValidateTextField('password', $errors);
+			if(sizeof($errors) == 0){
+				if(UserExists($_REQUEST['username'])){
+					if(GetUser($_REQUEST['username'])['password'] != $_REQUEST['password']){
+						$errors['match'] = 'incorrect password!';
 					}
+				}else{
+					$errors['DNE'] = 'user does not exist!';
 				}
-			</script>
+			}
+			if(sizeof($errors) == 0){
+				$_SESSION['username'] = $_REQUEST['username'];
+				header("Location: index.php");
+			}else{
+				if(isset($errors['username'])){
+					echo"<div class='required'>Please enter your username.</div>";
+				}
+				if(isset($errors['password'])){
+					echo"<div class='required'>Please enter your password.</div>";
+				}
+				if(isset($errors['match'])){
+					echo"<div class='required'>Your username and password do not match.</div>";
+				}
+				if(isset($errors['DNE'])){
+					echo"<div class='required'>There is no account associated with that username.</div>";
+				}
+			}
+		}
+		echo"<form method='post'>";
+		ShowTextField(true, 'username');
+		ShowPasswordField('password', 'password');
+		echo"<input type='submit' name = 'login' value='Login'>
+			</form>";
+	}
+
+	function IsLoggedIn(){
+		if(isset($_SESSION['username'])){
+			return true;
+		}
+		return false;
+	}
+
+	function PersonalHeading(){
+		echo"
+			<p class='personal'>hi ".$_SESSION['username']."</p>
 		";
+		Logout();
+	}
 
-		// if(isset($_REQUEST['delete'])){
-		// 	var_dump(GetPost($postID)['delKey']);
-		// 	echo"
-		// 		<script>
-		// 			alert('fdhldh');
-		// 			var key = prompt('Please enter delete key.');
-		// 			if(key == ".GetPost($postID)['delKey']."){
-		// 				".DeletePost($postID)."
-		// 			}
-		// 			else{
-		// 				alert('Wrong Delete Key! You do not have permission to delete this post.<br>');
-		// 			}
-		// 		</script>
-		// 	";
-		// }
-	}
-
-
-	//POST FUNCTIONS
-	function InsertBlogPost($author, $title, $body, $delKey){
+	function Logout(){
+		if(isset($_REQUEST['logout'])){
+			// echo"<script>
+			// 		alert('Are you sure you want to log out?')
+			// 	</script>
+			// ";
+			session_destroy();
+			header("Location: index.php");
+		}
+		echo"
+			<form>
+				<input class='personal' type='submit' name='logout' value='Log Out'>
+			</form>
+		";
+	}
+
+	function ValidDelete($postID){
+		if(isset($_SESSION['username'])){
```

Comment:
I'd suggest storing the user id in the session, not the username. It's best to use an autoincrementing int as your primary key, not a string. The posts table in your db should probably store a UserId, not a username. We can talk about this more if you'd like to discuss why this is the case. But generally speaking, every time one table links to another (a one-to-many or many-to-many relationship) it should be via an int.